File: app/src/main/java/com/keelim/arducon/MyApplication.java
package com.keelim.arducon;

import android.app.Application;
import android.content.Context;

import com.kakao.auth.KakaoSDK;
import com.keelim.arducon.adapters.KakaoSDKAdapter;

public class MyApplication extends Application {
    private static MyApplication instance;

    public static MyApplication getGlobalApplicationContext() {
        if (instance == null) {
            throw new IllegalStateException("This Application does not inherit com.kakao.MyApplication");
        }
        return instance;
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        // Kakao Sdk 초기화
        KakaoSDK.init(new KakaoSDKAdapter());
    }


    @Override
    public void onTerminate() {
        super.onTerminate();
        instance = null;
    }
}

File: app/src/main/java/com/keelim/arducon/MyApplication.kt
package com.keelim.arducon

import android.app.Application
import com.google.android.gms.ads.MobileAds
import com.keelim.arducon.utils.AppOpenManager
import timber.log.Timber

class MyApplication : Application() {
    private lateinit var appOpenManager: AppOpenManager
    override fun onCreate() {
        super.onCreate()

        MobileAds.initialize(this) { }
        appOpenManager = AppOpenManager(this) // 콜드 부팅에서 복귀시 ad*/

        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }

//        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
//        val themePref = sharedPreferences.getString("themePref", ThemeHelper.DEFAULT_MODE)
//        ThemeHelper.applyTheme(themePref!!)

//        applyTheme()
    }

//    private fun applyTheme(){
//        val migrationManger = MigrationManager(this)
//        GlobalScope.launch{
//            migrationManger.dataFlow.collect {
//                withContext(Dispatchers.Main){
//                    ThemeHelper.applyTheme(it)
//                }
//            }
//        }
//    }
}
